services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedTenantService.java

```diff
@@ -0,0 +1,630 @@
+                        TracingHelper.logError(span, String.format("tenant [%s] not found.", tenantId));
+                        return Future.succeededFuture(TenantResult.from(HttpURLConnection.HTTP_NOT_FOUND));
+                    }
+                    return Future.succeededFuture(TenantResult.from(
+                            HttpURLConnection.HTTP_OK,
+                            DeviceRegistryUtils.convertTenant(tenantId, successTenantRead.getPayload(), true),
+                            DeviceRegistryUtils.getCacheDirective(config.getCacheMaxAge())
+                    ));
+                });
+    }
+
+    /**
+     * Fetches a tenant by {@link X500Principal}.
+     *
+     * @param subjectDn The <em>subject DN</em> of the trusted CA certificate
+     *                  that has been configured for the tenant.
+     * @return Future TenantResult with {@code HttpURLConnection.HTTP_OK} if tenant found, {@code HttpURLConnection.HTTP_NOT_FOUND} else.
+     * @throws NullPointerException if {@code subjectDn} is {@code null}.
+     */
+    @Override
+    public Future<TenantResult<JsonObject>> get(final X500Principal subjectDn) {
+        Objects.requireNonNull(subjectDn);
+
+        return get(subjectDn, null);
+    }
+
+    /**
+     * Fetches a tenant by {@link X500Principal}.
+     *
+     * @param subjectDn The <em>subject DN</em> of the trusted CA certificate
+     *                  that has been configured for the tenant.
+     * @param span      The active OpenTracing span for this operation. It is not to be closed in this method!
+     *                  An implementation should log (error) events on this span and it may set tags and use this span as the
+     *                  parent for any spans created in this method.
+     * @return Future TenantResult with {@code HttpURLConnection.HTTP_OK} if tenant found, {@code HttpURLConnection.HTTP_NOT_FOUND} else.
+     * @throws NullPointerException if {@code subjectDn} is {@code null}.
+     */
+    @Override
+    public Future<TenantResult<JsonObject>> get(final X500Principal subjectDn, final Span span) {
+        Objects.requireNonNull(subjectDn);
+
+        return getForCertificateAuthority(subjectDn, span);
+
+    }
+
+    /**
+     * Finds an existing tenant by the subject DN of its configured certificate authority.
+     *
+     * @param subjectDn the subject DN to find the tenant with.
+     * @param span      The active OpenTracing span for this operation. It is not to be closed in this method!
+     *                  An implementation should log (error) events on this span and it may set tags and use this span as the
+     *                  parent for any spans created in this method.
+     * @return Future TenantResult with {@code HttpURLConnection.HTTP_OK} if tenant found, {@code HttpURLConnection.HTTP_NOT_FOUND} else.
+     */
+    private Future<TenantResult<JsonObject>> getForCertificateAuthority(final X500Principal subjectDn, final Span span) {
```

Comment:
Please follow the hono `javadoc` style and you could use `TenantManagementService` as a reference.